OutOfSchool/OutOfSchool.Common/Responses/ApiErrorsTypes.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Diagnostics;
+using System.Reflection;
+using OutOfSchool.Common.Responces;
+
+namespace OutOfSchool.Common.Responses;
+public abstract class ApiErrorsTypes
+{
+    private static string GetGroup()
+    {
+        StackTrace stackTrace = new StackTrace();
+        var frames = stackTrace.GetFrames();
+        MethodBase method;
+
+        try
+        {
+            method = frames[1].GetMethod();
+        }
+        catch
+        {
+            return "undefined";
+        }
```

Comment:
(Performance, readability) Please avoid exception-driven control flow. It's ususally non-obvious and bad from performance perspective. 
As for me (if I'm not missing something) you could just check if non-nullable `frame[1]` exists - this simple if will be faster than creating `catch` context and throwing/catching exception.

Ofc, the @strivitech comment is still valid and it's better to avoid using of reflection at all.